MoreLinq.Test/SliceTest.cs

```diff
@@ -125,5 +125,23 @@ public void TestSliceListOptimization()
             Assert.AreEqual(sliceCount, result.Count());
             Assert.IsTrue(result.SequenceEqual(Enumerable.Range(5, sliceCount)));
         }
+
+#if IREADONLY
+        /// <summary>
+        /// Verify that slice is optimized for <see cref="IReadOnlyList{T}"/> implementations and does not
+        /// unnecessarily traverse items outside of the slice region.
+        /// </summary>
+        [Test]
+        public void TestSliceReadOnlyListOptimization()
+        {
+            const int sliceStart = 4;
+            const int sliceCount = 3;
+            var sequence = new UnenumerableReadOnlyList<int>(new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
+            var result = sequence.Slice(sliceStart, sliceCount);
+
+            Assert.AreEqual(sliceCount, result.Count());
+            Assert.IsTrue(result.SequenceEqual(Enumerable.Range(5, sliceCount)));
```

Comment:
You can use `AssertSequenceEqual` here.

Reply:
I wanted to keep the tests as similar as possible for now, but I can edit them both.